packages/renderer/src/lib/image/PullImage.svelte

```diff
@@ -34,6 +38,27 @@ let selectedProviderConnection: ProviderContainerConnectionInfo | undefined;
 const lineNumberPerId = new Map<string, number>();
 let lineIndex = 0;
 
+async function resolveShortname(): Promise<void> {
+  if (selectedProviderConnection && selectedProviderConnection.type === 'podman') {
```

Comment:
Instead of having lots of if inside if statement, you could do return early pattern, IMHO it ease the reading

```ts
if (!selectedProviderConnection || selectedProviderConnection.type !== 'podman')
    return

// your logic
```